add-chat-push-notifications-auto-renew/SwiftPushTest/AppDelegate.swift

```diff
@@ -0,0 +1,267 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import UIKit
+import CryptoKit
+import UserNotifications
+import AzureCommunicationChat
+import AzureCommunicationCommon
+import BackgroundTasks
+
+@main
+class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterDelegate {
+
+    private (set) var appSettings = AppSettings()
+    private var tokenRefresher: ((@escaping (String?, Error?) -> Void) -> Void)?
+    private var userTokenClient: UserTokenClient!
+    private var chatClient: ChatClient?
+    var backgroundTask: UIBackgroundTaskIdentifier = .invalid
+    var notificationPresentationCompletionHandler: ((UNNotificationPresentationOptions) -> Void)?
+    var notificationResponseCompletionHandler: (() -> Void)?
+    
+
+    // MARK: App Launch
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        // Setup user token client and chat client
+        setupUserTokenClient()
+        setupChatClient()
+        
+        // Register for services
+        registerForPushNotifications()
+        registerBackgroundTasks()
+        
+        return true
+    }
+    
+    // MARK: - Setup Methods
+    
+    /// Sets up the UserTokenClient with token refresh logic.
+    private func setupUserTokenClient() {
+        userTokenClient = UserTokenClient(tokenIssuerURL: appSettings.tokenIssuerURL!)
+
+        tokenRefresher = { tokenCompletionHandler in
```

Comment:
Token refresher should return new token for given user_id. The current userTokenClient returns a new set of (user_id, token).